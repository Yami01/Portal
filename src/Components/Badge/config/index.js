/**
 *
 * @format
 * @flow
 *
 */

export const BADGE_SIZE = (size: number): number => {
  if (size > 0 && size) {
    return size
  } else {
    return 20
  }
}

export const BADGE_RADIUS = (size: number): number => {
  if (size > 0 && size) {
    return size / 2
  } else {
    return 10
  }
}
