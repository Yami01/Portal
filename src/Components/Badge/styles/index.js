/**
 *
 * @format
 * @flow
 *
 */
import { StyleSheet } from 'react-native'

import type { StylesType } from '../types'
import { Colors } from '@/Theme/Variables'

const BadgeStyles: StylesType = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.primary,
  },
  label: {
    fontSize: 10,
    color: Colors.white,
  },
})

export default BadgeStyles
