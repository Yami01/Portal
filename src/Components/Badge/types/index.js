/**
 *
 * @format
 * @flow
 *
 */

import type { Node } from 'react'
import type {
  ViewStyleProp,
  TextStyleProp,
} from 'react-native/Libraries/StyleSheet/StyleSheet'

export type PropsType = {
  children: Node,
  size?: number,
  extraStyle?: ViewStyleProp,
  labelStyle?: TextStyleProp,
  badgeRef?: () => void,
  testID?: string,
}

export type StylesType = {
  container: ViewStyleProp,
  label: TextStyleProp,
}
