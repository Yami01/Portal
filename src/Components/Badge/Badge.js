/**
 *
 * Badge
 * @format
 * @flow
 *
 */

import React, { memo } from 'react'
import type { Node, AbstractComponent } from 'react'
import { View } from 'react-native'

import type { PropsType } from './types'
import { BADGE_SIZE, BADGE_RADIUS } from './config'
import BadgeStyles from './styles'
import Text from '@/Components/Text'

const Badge: AbstractComponent<PropsType> = memo((props: PropsType): Node => {
  const {
    children,
    size = 20,
    extraStyle,
    labelStyle,
    badgeRef,
    testID,
  } = props
  const badgeSize = BADGE_SIZE(size || 0)
  const badgeRadius = BADGE_RADIUS(size || 0)

  return (
    <View
      style={[
        { width: badgeSize, height: badgeSize, borderRadius: badgeRadius },
        BadgeStyles.container,
        extraStyle,
      ]}
      ref={badgeRef}
      testID={testID}
    >
      <Text style={[BadgeStyles.label, labelStyle]}>{children}</Text>
    </View>
  )
})

export default Badge
