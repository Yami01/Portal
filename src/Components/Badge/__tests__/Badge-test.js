/**
 *
 * Badge Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'

import Badge from '../Badge'

describe('Badge', () => {
  const badgeNumber = 1
  test('Render Badge Notification', () => {
    const tree = renderer.create(<Badge>{badgeNumber}</Badge>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
