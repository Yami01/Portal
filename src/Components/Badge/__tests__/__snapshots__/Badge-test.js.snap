// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Render Badge Notification 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#E14032",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 10,
          },
          undefined,
        ],
      ]
    }
  >
    1
  </Text>
</View>
`;
